rules/static_test.go

```diff
@@ -36,255 +38,790 @@ var _ = Describe("Static", func() {
+
+						// Auto-allow IPIP traffic to other Calico hosts.
+						{
+							Match: Match().ProtocolNum(4).
+								DestIPSet("cali4-all-hosts").
+								SrcAddrType(AddrTypeLocal, false),
+							Action:  AcceptAction{},
+							Comment: "Allow IPIP packets to other Calico hosts",
+						},
+
+						// Non-workload traffic, send to host chains.
+						{Action: ClearMarkAction{Mark: 0xf0}},
+						{Action: JumpAction{Target: ChainDispatchToHostEndpoint}},
+						{
+							Match:   Match().MarkSet(0x10),
+							Action:  AcceptAction{},
+							Comment: "Host endpoint policy accepted packet.",
+						},
+					},
 				},
-			}))
-		})
-		It("IPv6: should include the expected workload-to-host chain in the filter chains", func() {
-			Expect(findChain(rr.StaticFilterTableChains(6), "cali-wl-to-host")).To(Equal(&Chain{
-				Name: "cali-wl-to-host",
-				Rules: []Rule{
-					{Match: Match().ProtocolNum(ProtoICMPv6).ICMPV6Type(130), Action: AcceptAction{}},
-					{Match: Match().ProtocolNum(ProtoICMPv6).ICMPV6Type(131), Action: AcceptAction{}},
-					{Match: Match().ProtocolNum(ProtoICMPv6).ICMPV6Type(132), Action: AcceptAction{}},
-					{Match: Match().ProtocolNum(ProtoICMPv6).ICMPV6Type(133), Action: AcceptAction{}},
-					{Match: Match().ProtocolNum(ProtoICMPv6).ICMPV6Type(135), Action: AcceptAction{}},
-					{Match: Match().ProtocolNum(ProtoICMPv6).ICMPV6Type(136), Action: AcceptAction{}},
-					{Action: JumpAction{Target: "cali-from-wl-dispatch"}},
-					{Action: ReturnAction{},
-						Comment: "Configured DefaultEndpointToHostAction"},
+				false: {
+					Name: "cali-OUTPUT",
+					Rules: []Rule{
+						// Untracked packets already matched in raw table.
+						{Match: Match().MarkSet(0x10),
+							Action: AcceptAction{}},
+
+						// To workload traffic.
+						{Match: Match().OutInterface("cali+").IPVSConnection(), Action: JumpAction{Target: "cali-to-wl-dispatch"}},
+						{Match: Match().OutInterface("cali+"), Action: ReturnAction{}},
+
+						// Auto-allow IPIP traffic to other Calico hosts.
+						{
+							Match: Match().ProtocolNum(4).
+								DestIPSet("cali4-all-hosts").
+								SrcAddrType(AddrTypeLocal, false),
+							Action:  AcceptAction{},
+							Comment: "Allow IPIP packets to other Calico hosts",
+						},
+
+						// Non-workload traffic, send to host chains.
+						{Action: ClearMarkAction{Mark: 0xf0}},
+						{Action: JumpAction{Target: ChainDispatchToHostEndpoint}},
+						{
+							Match:   Match().MarkSet(0x10),
+							Action:  AcceptAction{},
+							Comment: "Host endpoint policy accepted packet.",
+						},
+					},
 				},
-			}))
-		})
+			}
 
-		It("IPv4: Should return expected NAT prerouting chain", func() {
-			Expect(findChain(rr.StaticNATTableChains(4), "cali-PREROUTING")).To(Equal(&Chain{
-				Name: "cali-PREROUTING",
-				Rules: []Rule{
-					{Action: JumpAction{Target: "cali-fip-dnat"}},
-				}}))
+			// V6 should be unaffected.
+			expOutputChainIPIPV6 := map[bool]*Chain{
+				true: {
```

Comment:
I'm not keen on these true/false maps, it's not at all obvious what the true/false mean.  I suggest you create a separate named variable for each version

Reply:
Fixed.